api/v1/verticaautoscaler_webhook.go

```diff
@@ -206,8 +353,65 @@ func (v *VerticaAutoscaler) validateHPA(allErrs field.ErrorList) field.ErrorList
 		v.Spec.CustomAutoscaler.Hpa.Behavior.ScaleDown != nil && *v.Spec.CustomAutoscaler.Hpa.Behavior.ScaleDown.StabilizationWindowSeconds != 0 {
 		err := field.Invalid(pathPrefix.Child("hpa"),
 			v.Spec.CustomAutoscaler.Hpa,
-			"When scaledownThreshold is set, scaledown stabilization window must be 0")
+			"When scaleInThreshold is set, scalein stabilization window must be 0")
 		allErrs = append(allErrs, err)
 	}
 	return allErrs
 }
+
+// validateHPA will check if the HPA field is valid
+func (v *VerticaAutoscaler) validateScaleInThreshold(allErrs field.ErrorList) field.ErrorList {
+	// validate scaleInThreshold type
+	if v.HasScaleInThreshold() {
+		pathPrefix := field.NewPath("spec").Child("customAutoscaler")
+		for i := range v.Spec.CustomAutoscaler.Hpa.Metrics {
+			metric := &v.Spec.CustomAutoscaler.Hpa.Metrics[i]
+
+			if metric.Metric.Pods != nil && metric.ScaleInThreshold.Type != metric.Metric.Pods.Target.Type {
+				err := field.Invalid(pathPrefix.Child("hpa").Child("metrics").Index(i).Child("scaleInThreshold").Child("type"),
+					v.Spec.CustomAutoscaler.Hpa.Metrics[i].ScaleInThreshold.Type,
+					fmt.Sprintf("scaleInThreshold type %s must be of the same type as the threshold used for scale out %s",
+						metric.ScaleInThreshold.Type, metric.Metric.Pods.Target.Type),
+				)
+				allErrs = append(allErrs, err)
+			}
+
+			if metric.Metric.Object != nil && metric.ScaleInThreshold.Type != metric.Metric.Object.Target.Type {
+				err := field.Invalid(pathPrefix.Child("hpa").Child("metrics").Index(i).Child("scaleInThreshold").Child("type"),
+					v.Spec.CustomAutoscaler.Hpa.Metrics[i].ScaleInThreshold.Type,
+					fmt.Sprintf("scaleInThreshold type %s must be of the same type as the threshold used for scale out %s",
+						metric.ScaleInThreshold.Type, metric.Metric.Object.Target.Type),
+				)
+				allErrs = append(allErrs, err)
+			}
```

Comment:
There is a lot of code duplication. I suggest this:
```
metricTargets := map[string]*autoscalingv2.MetricTarget{
			"pods":              metric.Metric.Pods,
			"object":            metric.Metric.Object,
			"containerResource": metric.Metric.ContainerResource,
			"external":          metric.Metric.External,
			"resource":          metric.Metric.Resource,
		}
```
Then you can iterate over the map and make the checks.